codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/EndpointGenerator.java

```diff
@@ -134,6 +136,51 @@ public void run() {
                 });
     }
 
+    private void generateAwsEndpointResolverWrapper(GoWriter writer) {
+        Symbol awsEndpointResolver = SymbolUtils.createValueSymbolBuilder("EndpointResolver", AwsGoDependency.AWS_CORE)
+                .build();
+        Symbol resolverInterface = SymbolUtils.createValueSymbolBuilder(RESOLVER_INTERFACE_NAME).build();
+
+        Symbol wrappedResolverSymbol = SymbolUtils.createPointableSymbolBuilder("wrappedEndpointResolver").build();
+        writer.openBlock("type $T struct {", "}", wrappedResolverSymbol, () -> {
+            writer.write("awsResolver $T", awsEndpointResolver);
+            writer.write("resolver $T", resolverInterface);
+        });
+        writeExternalResolveEndpointImplementation(writer, wrappedResolverSymbol, "w", () -> {
+            writer.openBlock("if w.awsResolver == nil {", "}", () -> writer.write("goto fallback"));
+
+            writer.write("endpoint, err = w.awsResolver.ResolveEndpoint(ServiceID, region)");
+            writer.openBlock("if err == nil {", "}", () -> writer.write("return endpoint, nil"));
+            writer.write("");
+
+            writer.addUseImports(SmithyGoDependency.ERRORS);
+            writer.openBlock("if nf := (&$T{}); !errors.As(err, &nf) {", "}", SymbolUtils.createValueSymbolBuilder("EndpointNotFound",
+                    AwsGoDependency.AWS_CORE).build(), () -> writer.write("return endpoint, err"));
+            writer.write("");
```

Comment:
line length here probably throw checkstyle validation error. and little hard to read.

Reply:
It's under the max, but will change for readability.